syntax = "proto3";

package auth;

option go_package = "github.com/datapeice/astolfosplayer-backend/protos/gen/go/auth";

service AuthService {
    rpc Register (RegisterRequest) returns (RegisterResponse);
    rpc Login (LoginRequest) returns (LoginResponse);
}

message RegisterRequest {
    string username = 1;
    string password = 2;
    string security_key = 3;
}

message RegisterResponse {
    string token = 1;
}

message LoginRequest {
    string username = 1;
    string password = 2;
}

message LoginResponse {
    string token = 1;
}